bellbot_gui/src/gui2.py

```diff
@@ -0,0 +1,282 @@
+        try:
+            proxy = rospy.ServiceProxy('/bellbot_accept_target_single', NewTarget)
+            res = proxy(NewTargetRequest(""))
+            return res
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
+
+
+class GUI_Wait_For_Multi_Guests(object):
+    def __init__(self):
+        self.service_prefix = '/bellbot_gui_services'
+        self.buttons = [('Go', 'trigger_confirm_multi_guests')]
+        rospy.Service(self.service_prefix+'/trigger_confirm_multi_guests', std_srvs.srv.Empty, self.trigger_go)
+
+    def display(self, state=None):
+        buttons_content = strands_webserver.page_utils.generate_alert_button_page("", self.buttons, self.service_prefix)
+        message = "<div>Click to confirm that you would like to go to " + state.goal + "</div>" + "<div>" + state.text + "</div>"
+        www_content = message + buttons_content
+        strands_webserver.client_utils.display_content(display_no, www_content)
+
+    def trigger_go(self, req):
+        rospy.wait_for_service('/bellbot_accept_target_mult')
+        try:
+            proxy = rospy.ServiceProxy('/bellbot_accept_target_mult', NewTarget)
+            res = proxy(NewTargetRequest(""))
+            return res
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
+
+
+class GUI_Setup(object):
+    def __init__(self):
+        pass
+
+    def display(self, state=None):
+        strands_webserver.client_utils.display_url(display_no, random_page())
+
+
+class GUI_Operation_Feedback(object):
+    def __init__(self):
+        pass
+
+    def display(self, state):
+        #strands_webserver.client_utils.display_relative_page(display_no, 'cake.html')
+	    strands_webserver.client_utils.display_relative_page(display_no, 'livescreen.html')
+
+
+class GUI_User_Evaluation(object):
+    def __init__(self, http_root):
+        self.sub = None
+        self.http_root = http_root
+        self.feedback_filename = os.path.join(self.http_root, "user_feedback.csv")
+        # rospy.Service('/bellbot/gui/feedback_done', std_srvs.srv.Empty, self.handle_feedback_done)
+
+    def display(self, state=None):
+        if self.sub is not None:
+            self.sub.unregister()
+        strands_webserver.client_utils.display_relative_page(display_no, 'feedback.html')
+        self.sub = rospy.Subscriber("/bellbot_gui_feedback", String, self.cb_got_feedback)
+
+    def cb_got_feedback(self, data):
+        feedback = data.data
+        feedback = feedback.split(",")
+        for i in range(len(feedback)):
+            feedback[i] = int(feedback[i])
+        time_now_str = datetime.strftime(datetime.now(), '%Y-%m-%d %H:%M:%S')
+        line_lst = [time_now_str] + feedback
+        with open(self.feedback_filename, "a") as f:
+            writer = csv.writer(f, delimiter=",", quotechar='"', quoting=csv.QUOTE_NONNUMERIC)
+            writer.writerow(line_lst)
```

Comment:
As mentioned in the other file. This won't work due to missing permissions after install via debian package.